cpp/src/DataStorm/Contract.ice

```diff
@@ -391,6 +391,31 @@ module DataStormContract
         void s(long topicId, long elementId, DataSample sample);
     }
 
+    /// The reason for the session creation failure.
+    enum SessionError
+    {
+        // The session is already connected.
+        AlreadyConnected,
+
+        // Node is shutting down.
+        NodeShutdown,
+
+        // The connection failed
+        ConnectionFailure,
```

Comment:
Not obvious what this means. Is it more like a HTTP 502 error that a backend server can't be connected to or that an existing connection  has failed.

Reply:
Maybe we should have InternalError to group both, this and unknown